MOxUnit/moxunit_util_elem2str.m

```diff
@@ -18,20 +18,54 @@
 %     to capture the output from disp in a variable; hence only the class
 %     of the first input is shown
 %
-% NNO Jan 2014
+% NNO, SCL 2014-2015
 
     if nargin<2
         max_chars=100;
     end
 
-    if moxunit_util_platform_is_octave()
-        elem_str=sprintf('<%s>\n',class(elem));
-    else
-        % Octave does not support evalc
+    if ischar(elem)
+        % Strings are trivally supported by leaving them as-is
+        % We just encapsulate in quotation marks to make it clear they
+        % are a string.
+        elem_str=['''' elem ''''];
+
+    elseif isnumeric(elem) && numel(elem) < max_chars
+        % If the element is numeric, we can use mat2str
+        elem_str=mat2str(elem);
+
+    elseif exist('evalc', 'builtin')
+        % If we have `evalc`, we can just trust the `display` function
+        % provided by the environment to format the string correctly
+        % and capture it with `evalc`.
         elem_str=evalc('disp(elem)');
 
         % remove trailing newlines
         elem_str=elem_str(1:(end-2));
+
+        if iscell(elem)
+            % Encapsulate output in curly braces if it is a cell.
+            % These are not included by disp.
+            elem_str = ['{ ' strtrim(elem_str) ' }'];
```

Comment:
This gives strange output when, for example, `elem=cell([2,2,2])`.
It may be better to remove the special treatment of cells?